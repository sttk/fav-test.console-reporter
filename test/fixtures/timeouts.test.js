'use strict';

// Suite Level

describe('a suite of tests', function() {
  this.slow(100);
  this.timeout(500);

  it('should take less than 500ms', function(done) {
    setTimeout(done, 300);
  });

  it('should take less than 500ms as well', function(done) {
    setTimeout(done, 250);
  });
});

describe('a suite of tests (error)', function() {
  this.slow(100);
  this.timeout(200);

  it('should take less than 200ms', function(done) {
    setTimeout(done, 300);
  });

  it('should take less than 200ms as well', function(done) {
    setTimeout(done, 250);
  });
});

// Test level

it('should take less than 500ms', function(done) {
  this.slow(100);
  this.timeout(500);
  setTimeout(done, 300);
});
it('should take less than 200ms (error)', function(done) {
  this.slow(100);
  this.timeout(200);
  setTimeout(done, 300);
});

describe('a suite of tests (hook)', function() {
  this.slow(100);
  beforeEach(function(done) {
    this.timeout(3000);
    setTimeout(done, 2500);
  });

  it('test', function() {
  });
});

describe('a suite of tests (hook error)', function() {
  this.slow(100);
  beforeEach(function(done) {
    this.timeout(300);
    setTimeout(done, 500);
  });

  it('test', function() {
  });
});

