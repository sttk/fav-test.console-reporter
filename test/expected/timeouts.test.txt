

  a suite of tests
    ✓ should take less than 500ms (306ms)
    ✓ should take less than 500ms as well (250ms)

  a suite of tests (error)
    1) should take less than 200ms
    2) should take less than 200ms as well

  ✓ should take less than 500ms (304ms)
  3) should take less than 200ms (error)

  a suite of tests (hook)
    ✓ test

  a suite of tests (hook error)
    4) "before each" hook for "test"


  4 passing (4s)
  4 failing

  1) a suite of tests (error)
       should take less than 200ms:
     Error: Timeout of 200ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at listOnTimeout (internal/timers.js:531:17)
      at processTimers (internal/timers.js:475:7)

  2) a suite of tests (error)
       should take less than 200ms as well:
     Error: Timeout of 200ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at listOnTimeout (internal/timers.js:531:17)
      at processTimers (internal/timers.js:475:7)

  3) should take less than 200ms (error):
     Error: Timeout of 200ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at listOnTimeout (internal/timers.js:531:17)
      at processTimers (internal/timers.js:475:7)

  4) a suite of tests (hook error)
       "before each" hook for "test":
     Error: Timeout of 300ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at listOnTimeout (internal/timers.js:531:17)
      at processTimers (internal/timers.js:475:7)



